File: sdks/node/v1/test/import-smoke.test.ts
// Proves the generated client loads and exposes the surface consumers depend
// on. Cheap to run and catches a broken generation before the happy-path test
// has to stand up a mock server.
//
// These names are stable across regenerations: the API classes come from the
// spec's tags ("Interactions", "Authentication Details") and the models from
// the component schema names.

import { describe, expect, it } from 'vitest';

import {
  AuthenticationDetailsApi,
  BASE_PATH,
  Configuration,
  InteractionsApi,
  InteractionsBatchResultDtoFromJSON,
  InteractionsDtoToJSON,
  ResponseError,
} from '../src/index';

describe('generated client', () => {
  it('exposes the core runtime classes', () => {
    expect(Configuration).toBeTypeOf('function');
    expect(ResponseError).toBeTypeOf('function');
  });

  it('exposes the API classes', () => {
    expect(InteractionsApi).toBeTypeOf('function');
    expect(AuthenticationDetailsApi).toBeTypeOf('function');
  });

  it('exposes the model converters the happy path round-trips', () => {
    expect(InteractionsDtoToJSON).toBeTypeOf('function');
    expect(InteractionsBatchResultDtoFromJSON).toBeTypeOf('function');
  });

  it('defaults to the production host', () => {
    // typescript-fetch bakes in only the spec's first server, so unlike the
    // Ruby and Python clients no non-production host ships in this package.
    expect(BASE_PATH).toBe('https://api.movementinfrastructure.org');
  });

  it('takes a basePath override', () => {
    const config = new Configuration({ basePath: 'http://localhost:4010' });
    expect(config.basePath).toBe('http://localhost:4010');
  });
});

File: sdks/csharp/v1/test/Ddx.InteractionsApi.Test/ImportSmokeTest.cs
using System.Text.RegularExpressions;
using Ddx.InteractionsApi.Api;
using Ddx.InteractionsApi.Client;
using Ddx.InteractionsApi.Model;
using Xunit;

namespace InteractionsApiSdkTests;

/// <summary>
/// The generated assembly builds and exposes the surface consumers depend on.
/// Catches a broken generation before the happy-path test stands up a stub.
///
/// These names survive regeneration: API classes come from the spec's tags,
/// models from the component schema names.
/// </summary>
public class ImportSmokeTest
{
    [Fact]
    public void ExposesTheCoreClientTypes()
    {
        Assert.NotNull(typeof(ApiClient));
        Assert.NotNull(typeof(Configuration));
        Assert.NotNull(typeof(ApiException));
    }

    [Fact]
    public void ExposesTheApiClasses()
    {
        Assert.NotNull(typeof(InteractionsApi));
        Assert.NotNull(typeof(AuthenticationDetailsApi));
    }

    [Fact]
    public void ExposesTheModelsTheHappyPathRoundTrips()
    {
        Assert.NotNull(typeof(InteractionsDto));
        Assert.NotNull(typeof(InteractionsBatchResultDto));
        Assert.NotNull(typeof(WhoAmIApiKeyDto));
    }

    [Fact]
    public void ReportsTheVersionTheGeneratorConfigSet()
    {
        // Tracks packageVersion in openapi-generator-config.yaml.
        Assert.Matches(new Regex(@"^\d+\.\d+\.\d+"), Configuration.Version);
    }

    [Fact]
    public void DefaultsToTheProductionHost()
    {
        Assert.Equal("https://api.movementinfrastructure.org", new Configuration().BasePath);
    }

    [Fact]
    public void TakesABasePathOverride()
    {
        var config = new Configuration { BasePath = "http://localhost:4010" };
        Assert.Equal("http://localhost:4010", config.BasePath);
    }
}
